### PR TITLE
Support splitting build jobs into shards

### PR DESCRIPTION
**Pull request by nightkr**
Replaces #1, this is a generalization that supports sharding arbitrary combinations of build targets.

### CODE REVIEW

**Review by razvan — DISMISSED** [review]
lgtm!

**Review by razvan — DISMISSED** [review]
lgtm!

File: src/image_tools/args.py [hunk 1]
```diff
@@ -26,10 +26,13 @@ def bake_args() -> Namespace:
...
         help="Image version",
-        required=True,
+        default='0.0.0-dev',
         type=check_image_version_format,
     )
-    parser.add_argument("-p", "--product", help="Product to build images for")
+    parser.add_argument("-p", "--product",
+                        help="Product to build images for", action='append')
+    parser.add_argument("--shard-count", type=positive_int, default=1)
+    parser.add_argument("--shard-index", type=positive_int, default=0)
```
Comment (nightkr) [inline src/image_tools/args.py:35]: I'd copy the help text too.

File: src/image_tools/args.py [hunk 2]
```diff
@@ -53,7 +56,23 @@ def bake_args() -> Namespace:
         help="Image registry to publish to. Default: docker.stackable.tech",
         default="docker.stackable.tech",
     )
-    return parser.parse_args()
+    result = parser.parse_args()
+
+    if result.shard_index is not None and result.shard_index >= result.shard_count:
```
Comment (nightkr) [inline src/image_tools/args.py:61]:
```suggestion
    if result.shard_index >= result.shard_count:
```

**Review by razvan — APPROVED** [review]

**Pull request merged**